migrations/227_grading_method_split.sql

```diff
@@ -0,0 +1,33 @@
+-- Add new grading columns
+ALTER TABLE submissions 
+    ADD COLUMN IF NOT EXISTS grading_method_internal boolean DEFAULT True,
+    ADD COLUMN IF NOT EXISTS grading_method_external boolean DEFAULT False,
+    ADD COLUMN IF NOT EXISTS grading_method_manual boolean DEFAULT False;
+ALTER TABLE questions
+    ADD COLUMN IF NOT EXISTS grading_method_internal boolean DEFAULT True,
+    ADD COLUMN IF NOT EXISTS grading_method_external boolean DEFAULT False,
+    ADD COLUMN IF NOT EXISTS grading_method_manual boolean DEFAULT False;
+ALTER TABLE grading_jobs
+    ADD COLUMN IF NOT EXISTS grading_method_internal boolean DEFAULT True,
+    ADD COLUMN IF NOT EXISTS grading_method_external boolean DEFAULT False,
+    ADD COLUMN IF NOT EXISTS grading_method_manual boolean DEFAULT False;
+
+-- Update based on old values
+UPDATE submissions 
+    SET grading_method_internal = grading_method = 'Internal',
+        grading_method_external = grading_method = 'External',
+        grading_method_manual   = grading_method = 'Manual';
+UPDATE questions 
+    SET grading_method_internal = grading_method = 'Internal',
+        grading_method_external = grading_method = 'External',
+        grading_method_manual   = grading_method = 'Manual';
+UPDATE grading_jobs 
+    SET grading_method_internal = grading_method = 'Internal',
+        grading_method_external = grading_method = 'External',
+        grading_method_manual   = grading_method = 'Manual';
+
+-- Drop old grading method column
```

Comment:
We would like to eventually drop this. I wonder if we should wait until a successful test-run and after the migration though. This drop table could be a small PR on its own.

Reply:
The drop itself or the data migration? Again, I'm still somewhat paranoid if this proc is ever run twice on the DB... I've tried to sniff around with a better way to do the column update and then drop, but kinda unsure.

Unless what you're saying is just to do the column add/migration in this PR, and then the drop of the old columns/enums in another?